pennylane/labs/vibrational_ham/vibrational_class.py

```diff
@@ -0,0 +1,99 @@
+"""This module contains functions and classes to generate a pes object.
+This object stores all the necessary information to construct
+vibrational Hamiltonian for a given molecule."""
+
+import numpy as np
+import pennylane as qml
+import pyscf
+from pyscf.geomopt.geometric_solver import optimize
+from pyscf.hessian import thermo
+
+BOHR_TO_ANG = 0.529177
+
+
+def harmonic_analysis(scf_result, method="rhf"):
+    r"""Performs harmonic analysis by evaluating the Hessian using PySCF routines.
+    Args:
+       scf_result: pyscf object from electronic structure calculations.
+       method: Electronic structure method to define the level of theory
+            for harmonic analysis. Default is restricted Hartree-Fock 'rhf'.
+
+    Returns:
+       pyscf object containing information about the harmonic analysis.
+    """
+
+    method = method.strip().lower()
+    if method not in ["rhf", "uhf"]:
+        raise ValueError(f"Specified electronic structure method, {method} is not available.")
+
+    if method == "rhf":
+        hess = pyscf.hessian.rhf.Hessian(scf_result).kernel()
+    else:
+        hess = pyscf.hessian.uhf.Hessian(scf_result).kernel()
```

Comment:
```suggestion
hess = getattr(pyscf.hessian, method).Hessian(scf_result).kernel()
```

Might be more compact